runtime/src/server/middleware/types.ts

```diff
@@ -38,6 +39,10 @@ export type Props = {
 	[key: string]: any;
 };
 
+export interface Socket extends NetSocket {
+	encrypted?: boolean;
```

Comment:
I'm not sure if this is available on all versions of Node. I can't find it in the Node documentation and StackOverflow suggests it's not present in more recent versions (look at the comments): https://stackoverflow.com/a/10353248

It would be better to get any types that aren't Sapper-specific from DefinitelyTyped or the npm package itself. That would also have the benefit of helping us make sure it's available on the targeted platform

Reply:
@benmccann This is right that `req.connection` is getting deprecated in recent version of node. But, `req.socket` is available on all the versions of node > v0.3.0.
We are using `req.socket.encrypted` for the test of `https`. This is available of node v14 as well.
There are better ways to get protocol on most recent version of node like `req.protocol`. But, it's not available to earlier versions.
Let me know if you still have any concerns.

Reply:
Ok. Maybe we should install `@types/node` and use the `TLSSocket` type instead of making our own type though?

Reply:
Let me look into it.